src/Nethermind/Nethermind.Consensus/Producers/PayloadAttributes.cs

```diff
@@ -110,6 +120,16 @@ protected virtual int WritePayloadIdMembers(BlockHeader parentHeader, Span<byte>
             position += Keccak.Size;
         }
 
+        if (InclusionListTransactions is not null)
+        {
+            Transaction[] txs = InclusionListTransactions.Select(tx => Rlp.Decode<Transaction>(tx)).ToArray();
```

Comment:
I think we could avoid `ToArray` here and use `ToPooledList` - we might need to modify `PatriciaTree` to take `ReadonlySpan` or `IReadOnlyList` instead of array. Also create this collection only when Lenght > 0